lxd/storage/drivers/driver_btrfs_utils.go

```diff
@@ -484,6 +484,14 @@ func (d *btrfs) metadataHeader(vol Volume, snapshots []string) (*BTRFSMetaDataHe
 		return nil, err
 	}
 
+	// If vol is a snapshot itself, we need to fixup the metadata.
+	if vol.IsSnapshot() {
+		subVols[0].Readonly = false
```

Comment:
@stgraber this specific line makes me a little uncomfortable as it is reporting that the volume is writable when it isn't.

It may be a semantic issue of changing the definition of this function to return the state of the volume when it is migrated/restored elsewhere as opposed to what it is now.

Alternatively I may look at moving this special case up to where it is used, where we know that the main volume being sent is a snapshot. Either way as you say it is good enough to get 4.1 out.

Reply:
Yeah, we also must make sure we don't leak information. The target is not supposed to know the name of the snapshot on the source when a new container is created from that remote snapshot.